src/components/Popup/Popup.tsx

```diff
@@ -0,0 +1,200 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+import { Popper, PopperChildrenProps } from 'react-popper'
+import { Placement } from 'popper.js'
+import rtlCSSJS from 'rtl-css-js'
+
+import { childrenExist, customPropTypes, UIComponent, IRenderResultConfig } from '../../lib'
+import { ItemShorthand, Extendable } from '../../../types/utils'
+import { ComponentVariablesInput, IComponentPartStylesInput } from 'theme'
+import Portal from '../Portal'
+import PopupContent from './PopupContent'
+
+enum PositionParts {
+  top = 'top',
+  bottom = 'bottom',
+  left = 'left',
+  right = 'right',
+  before = 'before',
+  after = 'after',
+}
+
+type Position =
```

Comment:
Maybe it's just me, but I am still confused with the naming here. Let's discuss what the two parts of the position name mean in both RTL and LTR and agree on the naming in broader group.
I guess the main source of confusion is `start`/`end` meaning something different in combination with `top`/`bottom` vs `before`/`after`. I can see the inspiration in flexbox'es main/cross axis, but still...

Reply:
yea, I'll add this to your meeting notes to discuss it with everybody